### PR TITLE
Annotate plugins

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

- Annotate plugins. This makes all non-test code in GE null annotated.
- Rework some project properties to reduce duplication and increase correctness.

## Test methodology 

- Manual testing. CI will run tests.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs [hunk 1]
```diff
@@ -458,7 +473,7 @@ private Task<Stream> GetStreamFromHttpResponseAsync(HttpClient httpClient, Task<
                 return task.CompletedResult().Content.ReadAsStreamAsync();
             }
 
-            return null;
+            return null; // TODO NULLABLE returning a null Task is a no no
```
Comment (RussKie) [inline Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs:476]: Should this be `Task.FromResult(null)`?

File: Plugins/Gource/GourceStart.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ async Task DownloadImage((string email, string name) author)
                     var image = await AvatarService.DefaultProvider.GetAvatarAsync(author.email, author.name, imageSize: 90);
                     var filename = author.name + ".png";
 
-                    if (filename.IndexOfAny(Path.GetInvalidFileNameChars()) >= 0)
+                    if (image == null || filename.IndexOfAny(Path.GetInvalidFileNameChars()) >= 0)
```
Comment (RussKie) [inline Plugins/Gource/GourceStart.cs:119]:
```suggestion
                    if (image is null || filename.IndexOfAny(Path.GetInvalidFileNameChars()) >= 0)
```

File: Plugins/Statistics/GitStatistics/FormGitStatistics.cs [hunk 3]
```diff
@@ -82,7 +83,7 @@ public FormGitStatistics(IGitModule module, string codeFilePattern, bool countSu
             InitializeComplete();
         }
 
-        private void FormGitStatisticsSizeChanged(object sender, EventArgs e)
+        private void FormGitStatisticsSizeChanged(object? sender, EventArgs? e)
```
Comment (RussKie) [inline Plugins/Statistics/GitStatistics/FormGitStatistics.cs:86]: I don't think this is correct. Handlers generally have the following annotation: `(object? sender, EventArgs e)`

Comment (drewnoakes) [reply Plugins/Statistics/GitStatistics/FormGitStatistics.cs:86]: Agree this looks wrong. As before, I've just been annotating the current reality and fixing things up only when needed. I'll see if this is ready to change.

File: Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs (hunk 1 above)
Comment (drewnoakes) [reply Plugins/BuildServerIntegration/AppVeyorIntegration/AppVeyorAdapter.cs:476]: Looks like it. Requires some other changes.

**Review by RussKie — APPROVED** [review]
If it works - 👍

**Comment by drewnoakes** [comment]
Thanks @ivangrek

**Pull request merged**